hibernate-reactive-core/src/main/java/org/hibernate/reactive/adaptor/impl/ResultSetAdaptor.java

```diff
@@ -250,7 +266,123 @@ public String getCursorName() {
 
 	@Override
 	public ResultSetMetaData getMetaData() {
-		return null;
+		return new ResultSetMetaData() {
+			@Override
+			public int getColumnCount() {
+				return rows.columnsNames().size();
+			}
+
+			@Override
+			public boolean isAutoIncrement(int column) {
+				return false;
+			}
+
+			@Override
+			public boolean isCaseSensitive(int column) {
+				return false;
+			}
+
+			@Override
+			public boolean isSearchable(int column) {
+				return false;
+			}
+
+			@Override
+			public boolean isCurrency(int column) {
+				return false;
+			}
+
+			@Override
+			public int isNullable(int column) {
+				return columnNullableUnknown;
+			}
+
+			@Override
+			public boolean isSigned(int column) {
+				return false;
+			}
+
+			@Override
+			public int getColumnDisplaySize(int column) {
+				return 0;
+			}
+
+			@Override
+			public String getColumnLabel(int column) {
+				return rows.columnsNames().get(column-1);
+			}
+
+			@Override
+			public String getColumnName(int column) {
+				return rows.columnsNames().get(column-1);
+			}
+
+			@Override
+			public String getSchemaName(int column) {
+				return null;
+			}
+
+			@Override
+			public int getPrecision(int column) {
+				return 0;
+			}
+
+			@Override
+			public int getScale(int column) {
+				return 0;
+			}
+
+			@Override
+			public String getTableName(int column) {
+				return null;
+			}
+
+			@Override
+			public String getCatalogName(int column) {
+				return null;
+			}
+
+			@Override
+			public int getColumnType(int column) {
+				//TODO: we need SQLCLient to provide some column type info
+				return Types.VARCHAR;
```

Comment:
We need the `RowSet` to provide us with column type info if we want this to work really nicely without having to write a `@SqlResultSetMapping`.

Reply:
See https://github.com/eclipse-vertx/vertx-sql-client/issues/657